validator/src/journal/block_manager.rs

```diff
@@ -47,6 +47,41 @@ impl From<BlockStoreError> for BlockManagerError {
     }
 }
 
+/// An external block reference that is owned by and can be passed between validator components.
+/// Implements the `Drop` trait to allow for automatically decrementing the external reference
+/// count for the block ID in the block manager.
+pub struct BlockRef {
+    block_id: String,
+    block_manager: BlockManager,
+}
+
+impl BlockRef {
+    fn new(block_id: String, block_manager: BlockManager) -> Self {
+        BlockRef {
+            block_id,
+            block_manager,
+        }
+    }
+
+    pub fn block_id(&self) -> &str {
+        &self.block_id
+    }
+}
+
+impl Drop for BlockRef {
+    fn drop(&mut self) {
+        self.block_manager
+            .unref_block(self.block_id.as_str())
```

Comment:
I would perhaps add a comment on how this unref block will eventually not be callable outside of block ref.